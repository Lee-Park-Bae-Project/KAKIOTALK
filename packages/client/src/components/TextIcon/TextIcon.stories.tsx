import React from 'react';
import TextIcon from './index';

export default {
  title: 'Component/TextIcon',
  component: TextIcon,
};

export const Default = () => (
    <TextIcon icon='Close' size='2rem' iconPosition='left' text='close' color='white'/>
);

export const Right = () => (
  <TextIcon icon='Close' size='2rem' iconPosition='right' text='close'/>
);

export const Top = () => (
  <TextIcon icon='Close' size='2rem' iconPosition='top' text='close'/>
);

export const Bottom = () => (
  <TextIcon icon='Close' size='2rem' iconPosition='bottom' text='close'/>
);

export const CustomColor = () => (
  <TextIcon icon='Close' size='2rem' iconPosition='bottom' text='close' color='red' textColor='blue'/>
);
