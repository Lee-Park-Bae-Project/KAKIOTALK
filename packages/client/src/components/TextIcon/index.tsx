import React, {
  FC, ReactElement,
} from 'react'
import styled from 'styled-components'
import Icon, { IconType } from 'Icon/Icon'

type iconPositionType = 'left' | 'right' | 'top' | 'bottom'

interface ContainerProp {
  /** text에 대한 icon의 상대적 위치 */
  iconPosition?: iconPositionType
  /** 클릭핸들러 */
  onClick?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void
}
interface TextProp {
  /** 아이콘과 함께 표시될 텍스트 */
  text: string
  /** 텍스트 색 지정할 때 이용 */
  textColor?: string
  /** 텍스트 크기 지정할 떄 이용 */
  textSize?: string
}

interface Prop extends TextProp, ContainerProp {
  /** icon 이름 */
  icon: IconType
  /** 색을 지정할 때 이용 */
  color?: string
  /** 크기 지정할 때 이용 */
  size?: string | number
  /** 구글프로필사진 URL */
  imageUrl?: string
  /** 팝업할 프로필 컴포넌트 */
  children?: ReactElement | boolean
  /** 상태메시지 */
  statusMessage?: string
}

const directionMap = {
  top: 'column',
  bottom: 'column-reverse',
  left: 'row',
  right: 'row-reverse',
}

const S = {
  Container: styled.div<ContainerProp>`
    display: flex;
    flex-direction: ${(props) => {
    if (!props.iconPosition) {
      return 'left'
    }
    return directionMap[props.iconPosition]
  }};
    width: fit-content;
    height: fit-content;
    align-items: center;
    justify-content: center;
  `,
  TextWrapper: styled.span<TextProp>`
    color: ${(props) => props.textColor};
    line-height: 1.5rem;
    padding: 1rem;
    font-size: ${(props) => props.textSize};
  `,
  Image: styled.img`
    width: 2.5rem;
    height: 2.5rem;
    border-radius: 40%;
    cursor: pointer;
  `,
  statusMessageWrapper: styled.div`
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    width: 12rem;
    color: #BDBDBD;
  `,
}

/**
 * text와 icon 함께 사용할 때 이용
 */
const TextIcon: FC<Prop> = ({
  icon,
  color,
  size,
  iconPosition = 'left',
  onClick = undefined,
  text,
  textColor = 'black',
  textSize = '1rem',
  imageUrl = null,
  children,
  statusMessage = null,
}) => (
  <S.Container iconPosition={iconPosition}>
    {imageUrl ? (
      <S.Image src={imageUrl} onClick={onClick} />
    ) : (
      <Icon icon={icon} color={color} size={size} onClick={onClick} />
    )}
    {children && children}
    <S.TextWrapper text={text} textColor={textColor} textSize={textSize}>
      <strong>{text}</strong>
      {statusMessage && (
        <S.statusMessageWrapper>{statusMessage}</S.statusMessageWrapper>
      )}
    </S.TextWrapper>
  </S.Container>
)

export default TextIcon
