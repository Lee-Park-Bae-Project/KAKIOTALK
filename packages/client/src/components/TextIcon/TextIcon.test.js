/* eslint-disable no-undef */
import React from 'react'
import {
  mount, shallow,
} from 'enzyme'
import TextIcon from 'components/TextIcon'

describe('<TextIcon/>', () => {
  it('matches snapshot', () => {
    const wrapper = mount(<TextIcon icon='Account' text='hi'/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('matches props', () => {
    const wrapper = mount(<TextIcon icon='Account' text='hihi'/>)
    expect(wrapper.props().icon).toBe('Account')
    expect(wrapper.props().text).toBe('hihi')
  })
})
