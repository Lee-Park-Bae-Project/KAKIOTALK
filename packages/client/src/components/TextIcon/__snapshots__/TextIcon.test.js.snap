// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextIcon/> matches snapshot 1`] = `
<TextIcon
  icon="Account"
  text="hi"
>
  <styled.div
    iconPosition="left"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "hSFraI",
            "rules": Array [
              "
    display: flex;
    flex-direction: ",
              [Function],
              ";
    width: fit-content;
    height: fit-content;
    align-items: center;
    justify-content: center;
  ",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      iconPosition="left"
    >
      <div
        className="sc-bwzfXH hSFraI"
      >
        <Icon
          icon="Account"
          size="2rem"
        >
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "NKuto",
                    "rules": Array [
                      "
    cursor: pointer;
    display: flex;
  ",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="sc-bdVaJa NKuto"
              >
                <ForwardRef(SvgAccount)
                  fill="currentColor"
                  height="2rem"
                  width="2rem"
                >
                  <svg
                    fill="currentColor"
                    height="2rem"
                    width="2rem"
                  >
                    account.svg
                  </svg>
                </ForwardRef(SvgAccount)>
              </div>
            </StyledComponent>
          </styled.div>
        </Icon>
        <styled.span
          text="hi"
          textColor="black"
          textSize="1rem"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": false,
                  "lastClassName": "gssNWV",
                  "rules": Array [
                    "
    color: ",
                    [Function],
                    ";
    line-height: 1.5rem;
    padding: 0.5rem;
    font-size: ",
                    [Function],
                    ";
  ",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            text="hi"
            textColor="black"
            textSize="1rem"
          >
            <span
              className="sc-htpNat gssNWV"
            >
              hi
            </span>
          </StyledComponent>
        </styled.span>
      </div>
    </StyledComponent>
  </styled.div>
</TextIcon>
`;
